stripecardscan/src/main/java/com/stripe/android/stripecardscan/cardimageverification/CardImageVerificationActivity.kt

```diff
@@ -166,15 +175,29 @@ internal open class CardImageVerificationActivity :
 
             override fun cardReadyForVerification(pan: String, frames: Collection<SavedFrame>) {
                 launch {
-                    when (
-                        val result = uploadSavedFrames(
-                            stripePublishableKey = params.stripePublishableKey,
-                            civId = params.cardImageVerificationIntentId,
-                            civSecret = params.cardImageVerificationIntentSecret,
-                            savedFrames = frames,
-                            imageConfigs = imageConfigs,
-                        )
-                    ) {
+                    mainLoopStatsTracker?.trackResult("complete")
+                    mainLoopStatsTracker = null
+
+                    val imageCompressionStat = Stats.trackTask("image_compression_duration")
+
+                    val verificationFramesAndPayload = frames.toVerificationFrameData(imageConfigs)
+                    currentScanPayloadInfo = verificationFramesAndPayload.second
+
+                    imageCompressionStat.trackResult("complete")
```

Comment:
is there any way to move this into the `uploadSavedFrames` where it was before? we should be able to access stats from there.

Reply:
Sorry? This is a new stat specifically for tracking the length of time to encode/compress the Bitmaps. Essentially we wanted to see if JPEG was faster or slower than HEIC/WEBP. The thesis being, we might have reduced the payload size, but did we increase the completion duration and thus more ppl will fall off.